clients/cobol-lsp-vscode-extension/src/services/copybook/CopybookDownloadService.ts

```diff
@@ -157,9 +177,10 @@ export class CopybookDownloadService implements vscode.Disposable {
                                 " copybook(s) left.",
                         });
                     }
-                    await CopybookDownloadService.handleCopybook(dataset, cp, errors, isUSS);
+                    promises.push(CopybookDownloadService.handleCopybook(progress, dataset, cp, errors, isUSS));
                 }
             }
+            await Promise.all(promises);
```

Comment:
This is great, but a little naive. As long as the service can handle many parallel requests it will work well. But we may also run into a situation where we would have to _throttle_ the amount of parallel requests in order not to overwhelm the remote server.

Probably not worth extra work at the moment, but worth keeping in mind in case we would get error reported in this area.